dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -2041,21 +2041,37 @@ boolean reportErrors(Result result, Collection<String> restrictedAccessAttempts,
         List<String> unresolvedVersionless = new ArrayList<>();
         for (Map.Entry<String, String> versionlessResolved : result.getVersionlessFeatures().entrySet()) {
             if(versionlessResolved.getValue() == null){
-                String platformBaseName = featureRepository.getPlatformForVersionlessFeature(versionlessResolved.getKey());
-
+                Set<String> platforms = featureRepository.getPlatformsForVersionlessFeature(versionlessResolved.getKey());
                 boolean compatibleWithPlatform = true;
-                if(platformBaseName != null && !result.getResolvedPlatforms().isEmpty()){
-                    for(String resolvedPlat : result.getResolvedPlatforms()){
-                        if(resolvedPlat.toLowerCase().startsWith(platformBaseName.toLowerCase()) && 
-                            featureRepository.getVersionlessFeatureVersionForPlatform(versionlessResolved.getKey(), resolvedPlat) == null){
 
+                //get the compatibility feature base name tied to this versionless feature
+                ProvisioningFeatureDefinition compatibility = null;
+                if(platforms != null && !platforms.isEmpty()){
+                    for(String platform : platforms){
+                        compatibility = featureRepository.getCompatibilityFeature(platform);
+                        if(compatibility != null){
+                            break;
+                        }
+                    }
+                }
+                if(compatibility == null){
+                    break;
+                }
+                compatibility = featureRepository.getCompatibilityFeature(platforms.toArray()[0].toString());
```

Comment:
This line worries me - I did add some null checks over the weekend - should we add here?